addons/crm/wizard/crm_merge_opportunities.py

```diff
@@ -37,12 +37,23 @@ def default_get(self, fields):
     team_id = fields.Many2one(
         'crm.team', 'Sales Team',
         compute='_compute_team_id', readonly=False, store=True)
+    opportunity_company_ids = fields.One2many('res.company', compute="_compute_opportunity_company_ids")
```

Comment:
Meh ... don't think it is a good idea to touch the merge wizard by adding MC stuff...

Reply:
At the moment, all users are available to choose from. When the destination lead's company is not in the user's allowed companies, it raises an inconsistency error for the record.

At first I used intersection of companies, but it is too restrictive as only users having access to all companies could be selected.

Then in this version I keep the union of companies. But again, this could lead to the error described above.

The issue is that in order to be correct, we would have to "compute" the destination lead's company based on the list of merged leads, and restrict the users accordingly, and this feels like a bad idea.

It looks like it is a Deadlock situation :/

Reply:
Actually we thought of forcing the merge to be in the same company and avoid MC issues when merging document, notably because you may have links with sale orders belonging to yet another company for example. We have some tasks for that
* https://www.odoo.com/web#id=2951093&cids=1&menu_id=6444&action=5050&model=project.task&view_type=form that is about relaxing MC rules as they are quite annoying
* https://www.odoo.com/web#id=2955805&cids=1&menu_id=6444&action=5050&model=project.task&view_type=form that is about MC merge, and is waiting some brain time on my side

This change is a bit transversal as it limit only displayed users, but I would make a MC update of crm in a single task to be sure we handle all situations. So I would not do it here and move it into a dedicated sub-task on a main crm MC task ^^ .